lib/giraph/remote/response.rb

```diff
@@ -0,0 +1,9 @@
+module Giraph
+  module Remote
+    # Dummy Hash-wrapper to enable precise `instance_of?` checks
+    # Allows us to differentiate JSON responses that are
+    # coming from remote GraphQL interface vs regular Hash objects
+    class Response < Hash
```

Comment:
Good suggestion but it doesn't work the same way (cannot be identified recursively):

```rb
raw_json = {a:1, b:[2,3], c: {d:"d", e: 12.2}}.to_json
module Foo; end
class FooHash < Hash; end

# Marker
hash = JSON.parse(raw_json, symbolize_names: true).extend(Foo)
=> {:a=>1, :b=>[2, 3], :c=>{:d=>"d", :e=>12.2}}
hash.is_a? Foo
=> true
hash[:c].is_a? Foo
=> false

# Subclass
hash = JSON.parse(raw_json, symbolize_names: true, object_class: FooHash)
=> {:a=>1, :b=>[2, 3], :c=>{:d=>"d", :e=>12.2}}
hash.is_a? FooHash
=> true
hash[:c].is_a? FooHash
=> true
```